src/shared/components/msktable/MSKTable.tsx

```diff
@@ -0,0 +1,110 @@
+import {Column as SimpleTableColumn, Row as SimpleTableRow, default as SimpleTable} from "../simpleTable/SimpleTable";
+import * as React from 'react';
+import {observable, computed, action} from "mobx";
+import {observer} from "mobx-react";
+
+export type Column<T> = {
+    name: string;
+    filter?:(data:T, filterString:string)=>boolean;
+    sort?:(a:T, b:T)=>number;
+    render:(data:T)=>JSX.Element;
+};
+
+type MSKTableProps<T> = {
+    columns:Column<T>[];
+    data:T[];
+};
+
+class MSKTableStore<T> {
+    @observable public filterString:string;
+    @observable public page:number;
+    @observable public itemsPerPage:number;
+    @observable public sortColumn:string;
+    @observable public sortAscending:boolean;
+    @observable public columns:Column<T>[];
+    @observable.ref public data:T[];
+    @computed get sortedFilteredData():T[] {
+        let filtered:T[];
+        if (this.filterString) {
+            filtered = this.data.filter((datum:T)=>{
+                let match = false;
+                for (let i=0; i<this.columns.length; i++) {
```

Comment:
this is the inner loop.  gonna happen a lot so, worth optimizing.  you could cache this.columns.length in local variable.